Версия 3.1

Medvedina opened the pull request:
- Изменён алгоритм логгирования, теперь используется стандартный логгер Python
- Убраны лишние проверки в service_functions.calc
- Повторяющийся код заменён на функции
- Изменены тесты
- Добавлен .gitignore с файлами, которые не должны попадать в commit
- Почищен репозиторий от лишних файлов и "продуктов жизнедеятельности программы"
- Переменные переименованы по PEP8
- Перенесены импорты, немного упрощён код
- Ошибки фиксируются через raise, а не возвращением строки
- Добавлен графический интерфейс (даже 2)
- Добавлен калькулятор IP и тесты к нему

Medvedina committed:
chore(.pylintrc) - удалён файл .pylintrc

Medvedina committed:
chore(history) - удален __pycache__

Medvedina committed:
chore(history): удалён logs.txt

Medvedina committed:
chore(service): Удалён __pycache__

Medvedina committed:
chore(tests): Удалён __pycache__

Medvedina committed:
fix(calc.py): исправление ошибок

- Добавлена точка входа
- Импорты выполняются раньше
- Переименованы переменные в соотв. с PEP8

Medvedina committed:
feat(logs_script.py): изменён принцип логгирования

- Теперь используется стандартный логгер языка Python, устраняющий возможные ошибки и добавляющий возможность расширения функционала

Medvedina committed:
fix(service_functions.py): внедрение нового алгоритма логгирования, исправление ошибок

- Изменён алгоритм логгирования, применяется стандартный логгер python
- Удалены лишние проверки
- Импорты выполняются раньше
- Повторяющийся код заменён функциями
- Переименованы переменные в соотв. с PEP8

Medvedina committed:
test(test_calc.py): изменены тесты для функции calc в соотв. с новыми проверками

Medvedina committed:
feat(.gitignore): добавлен .gitignore

Medvedina committed:
fix(logs_script.py) - исправлена ошибка, ломающая всё...

Вместо debug стало DEBUG

Medvedina committed:
Merge branch 'main' into develop

Medvedina committed:
feat(errors): изменение способа перехвата ошибок

Medvedina committed:
feat(service_functions.py): изменения в учёте ошибок

Medvedina committed:
feat(calc.py): изменён метод работы с ошибками

Medvedina committed:
Add files via upload

Medvedina committed:
Merge pull request #7 from Medvedina/errors

Функционал перехвата ошибок

Medvedina committed:
технический сейв

Medvedina committed:
fix(service_functions.py): Изменена функция rand

Теперь функция принимает 3 значения, вместо двух. Сделано для функционирования графической оболочки

Medvedina committed:
fix(test_rand): измен набор тестов под новую функцию

Medvedina committed:
feat(graphic_calc.py): добавлен графический интерфейс

Medvedina committed:
chore(calc.py): удален консольный интерфейс за ненадобностью

Medvedina committed:
refactor(readme.md): изменено описание

Medvedina committed:
feat(beautiful_calc.py): улучшеная графика

Библиотека tkinter заменена на более приятный глазу custom_tkinter

Medvedina committed:
feat(requirements.txt): добавлен customtkinter

Medvedina committed:
feat(beautiful_calc.py): добавлен калькулятор IP

Добавлен калькулятор IP адресов.
По ip и маске выводит:
- Двоичный IP
- Адрес сети
- Индекс маски подсети
- Число доступных хостов

Medvedina committed:
feat(service_functions.py): Добавлен класс IP

Класс IP содержит все атрибуты и функции, необходимые для калькулятора ip.
Также:
Удалена ф-ция mode_check() за ненадобностью (использовалась для консольного интерфейса)

Medvedina committed:
fix(errors.py): добавлен класс PrefixError

Ошибка, которая может встретиться в калькуляторе IP

Medvedina committed:
tests(tests/test_modecheck.py) - Удалён за ненадобностью

Medvedina committed:
tests(test_ipcalc.py): Тесты для калькулятора IP

4 позитивных, 15 негативных тестов, покрывающих все возможные запросы пользователя

Medvedina committed:
docks(readme.md): изменено описание, добавлены скриншоты

Medvedina committed:
fix(python_tests.yml): добавлена активация теста калькулятора ip

Также убран test_modecheck.py

Medvedina committed:
feat(beautiful_calc): добавлена тёмная тема и селектор выбора темы

Medvedina committed:
fix(beautiful_calc.py): исправление визуала

Передвинуто уведомление об успешной очистке логов, теперь оно не будет закрывать меню выбора темы

Medvedina committed:
docks(README.md): Добавлены скриншоты

Medvedina committed:
feat(beautiful_calc.py): Добавлена тёмная тема

- Добавлено меню выбора темы (светлая, тёмная)
- Изменены цвета в светлой теме
- В калькуляторе IP CheckBox заменён на Option Menu для удобства

Medvedina committed:
feat(beautiful_calc.py): Добавлен перевод в разные системы счисления

- Добавлен функционал перевода чисел в различные системы счисления
- Изменена функция mode_ipcalc(). Больше не используется глобальная переменная

Medvedina committed:
feat(service_functions.py): Перевод систем счисления

- Добавлена функция numsys(original_system, number), отвечающая за функционал перевода систем счисления
- Изменены функции класса IP, а именно некоторые условия

Medvedina committed:
tests(test_ipcalc.py): Изменены проверки под новые условия

Medvedina committed:
fix(service_functions.py): изменён перевод из двоичной системы

- Убран цикл при переводе, исправлена ошибка, когда в строке "Двоичная форма" писалась десятичная

Medvedina committed:
tests(test_numsys.py): Добавлен тест для нового функционала

Medvedina committed:
docks(README.md): Изменено описание, добавлены скриншоты

Medvedina committed:
tests(python_tests.yml): Активация нового теста

Medvedina committed:
Merge branch 'main' into develop

Medvedina reviewed (commented):
Случайно изменил yml не в той ветке

shekshuev commented:
Опять же коммитов много, по ним бегать неудобно. Опишу в целом по ветке develop
`beautiful_calc.py` и `graphic_calc.py`
1. Зачем ты определяешь функции в функциях? Если ты хочешь некую инкапсуляцию, посмотри в сторону ООП. Сделай класс с формами, кнопками, окошками и пр., и все функции делай как методы класса

`service_functions.py`
1. зачем ты возвращаешь print
```python
def notify_user(*args):
    return print(*args)
```
2. Этот код
```python
def random_check(test_value):
    if (not test_value.isnumeric() and test_value != ' ' and
            test_value != '' and test_value != '-'):
        return True
    else:
        return False
```
можно проще, типа такого, оптимизируй только
```python
def random_check(test_value):
    return not (not test_value.isnumeric() and test_value != ' ' and
            test_value != '' and test_value != '-')
```
не принято на `if` возвращать `True` / `False`. У тебя же в `if` логическое выражение, вот и возвращай его

3. 31-32 строки, наверно логгера не хватает
4. log_script.py 
```python
logger.setLevel(logging.DEBUG)
```
этот уровень нужень только во время отладки. Условно когда твое приложение будет работать у пользователя, уровеь должен быть выше. Подумай, как его считать динамически, например через конфиг

По тестам:
1. Тестовое покрытие оцени сам
2. Структура тестов должна быть примерно такая
```
├─ testing/
├─ data/
│  ├─ mock.py
│  ├─ constants.py
│  ├─ ... 
├─ tests/
│  ├─ test_....py
│  ├─ ...
├─ conftest.py
├─ pytest.ini
```
В файле mock ты помещаешь моковые данные, например
```python
test_cases = [
    ("12 + 15 / (3 + 2) * 4", 12 + 15 / (3 + 2) * 4),
    ("(2 + 3) * 5", (2 + 3) * 5),
    ("(8 / 4) + (3 * 2)", (8 / 4) + (3 * 2)),
    ("(10 + 2) / 3", (10 + 2) / 3),
    ("3 * (4 + 2) - 5", 3 * (4 + 2) - 5),
    ("test", er.InputError),
    ("(32+3", er.BracketError),
    ("12 / 0", er.DivZero),
    ("", er.InputError),
    ("((((31*12)-15/3)+2^3)-5)", ((((31*12)-15/3)+2**3)-5)),
    ("-", er.InputError)
]
```
В constants константы, например тексты ошибок, которые ты ожидаешь
в папке tests непосредственно твои тесты
`conftest.py` содержит конфигурацию тестов. Тут нужно указывать фикстуры и все, что с ними связано
`pytest.ini` при необходимости, это конфиг

3. Импорты должны быть на уровне файла, а не в функциях (`test_calc.py`, 24 строка)
4. Тесты должны быть однозначные, не должно быть в них условий
```python
if expected in errors:
        with pytest.raises(expected):
            calc(expression)
    else:
        assert calc(expression) == expected
```
В данном случае ты смешал позитивные и негативные кейсы, необходимо их разделить

В принципе, по всем тестам одинаковые замечания. Поправь.
Очень хорошо, что ты используешь параметризацию. 

Ты протестировал только уровень сервиса. Оптимально сделать такую структуру
```
├─ testing/
├─ data/
│  ├─ mock.py
│  ├─ constants.py
│  ├─ ... 
├─ tests/
│  ├─ test_service.py
│  ├─ test_beautiful_calc.py
│  ├─ test_graphic_calc.py
├─ conftest.py
├─ pytest.ini
```

Обрати внимание, что юнит тестами ты тестируешь только части приложения. Если ты тестируешь UI, правильность рассчетов тебе проверять не нужно, она тестируется в `test_services.py`. Тесты GUI должны включать проверку состояний UI элементов (кнопка заблокировалас.разблокировалась, label имеет корректный текст, окно открылось / закрылось и пр.)

Прогресс очень хороший у тебя, молодец!

vadimnet commented:
Тип калькулятор написать не может...

Medvedina committed:
fix(service_functions.py): Исправления ошибок, удаление лишнего кода

- Удалена функция notify_user() (осталась от консольного интерфейса)
- Исправлена неработающая проверка правильности ввода в восьмеричной системе
- Исправлена проверка random_check (упрощён код)
